### PR TITLE
Configure http pools via mongoose_wpool in tests

### PR DESCRIPTION
**Pull request by michalwski**
Proposed changes include:
* start http pools via `mongoose_wpool`
* update doc in places where the old http pool starting was used

### CODE REVIEW

File: doc/advanced-configuration/outgoing-connections.md [hunk 1]
```diff
@@ -54,6 +54,38 @@ will be expanded to the following configuration:
 {redis, <<"d.eu">>, default, PoolOpts, ConnOpts}
 ```
 
+## HTTP connections setup
+
+Some MongooseIM modules needs HTTP connection to external service.
```
Comment (goddammit) [inline doc/advanced-configuration/outgoing-connections.md]: Some MongooseIM modules need an HTTP connection to external services.

File: doc/advanced-configuration/outgoing-connections.md [hunk 2]
```diff
@@ -54,6 +54,38 @@ will be expanded to the following configuration:
...
+These pools needs to be configured and started before the module needs them.
```
Comment (goddammit) [inline doc/advanced-configuration/outgoing-connections.md]: These pools need to be configured and started before the module needs them.

File: doc/advanced-configuration/outgoing-connections.md [hunk 3]
```diff
@@ -54,6 +54,38 @@ will be expanded to the following configuration:
...
+Sample configuration looks like the following:
```
Comment (goddammit) [inline doc/advanced-configuration/outgoing-connections.md]: Below is a sample configuration:
See a sample configuration below:

File: doc/advanced-configuration/outgoing-connections.md [hunk 4]
```diff
@@ -54,6 +54,38 @@ will be expanded to the following configuration:
...
+
+```erlang
+{outgoing_pools, [
+ {http, global, default, PoolOptions, ConnectionOptions}
+]}.
+```
+
+`PoolOptions` are described above, below there are described recommended `PoolOptions` for `HTTP` pools:
```
Comment (goddammit) [inline doc/advanced-configuration/outgoing-connections.md]: PoolOptions` are described above, below there are the recommended `PoolOptions` for `HTTP` pools:

File: doc/advanced-configuration/outgoing-connections.md [hunk 5]
```diff
@@ -54,6 +54,38 @@ will be expanded to the following configuration:
...
+
+* `strategy` - the recommended value is `available_worker`
+* `call_timeout` - it should be equal or longer than the value set in `request_timeout` below.
+
+`ConnectionOptions` can take following `{key, value}` pairs:
```
Comment (goddammit) [inline doc/advanced-configuration/outgoing-connections.md]: `ConnectionOptions` can take the following `{key, value}` pairs:

File: doc/modules/mod_push_service_mongoosepush.md [hunk 6]
```diff
@@ -1,39 +1,41 @@
...
- * `service` - has to be specified and the value must be valid and supported by the MongoosePush push service provider. 
+ * `service` - has to be specified and the value must be valid and supported by the MongoosePush push service provider.
   E.g. `fcm`, `apns`.
  * `device_id` - has to be specified and the value must be valid device token received from push notification service provider specified in `service` option
- 
-In addition to those `publish-options` you may specify also `silent` option, that when set to 
-`true` will result in "silent" notification. 
-Silent notifications send only data payload to push notifications service provider with all 
+
+In addition to those `publish-options` you may specify also `silent` option, that when set to
```
Comment (goddammit) [inline doc/modules/mod_push_service_mongoosepush.md]: In addition to those `publish-options` you may also specify a `silent` option, which, when set to `true`, will result in a "silent" notification.

File: doc/modules/mod_push_service_mongoosepush.md [hunk 7]
```diff
@@ -1,39 +1,41 @@
...
+`true` will result in "silent" notification.
+Silent notifications send only data payload to push notifications service provider with all
```
Comment (goddammit) [inline doc/modules/mod_push_service_mongoosepush.md]: Silent notifications only send data payload to push notifications' service provider with all fields specified in the notification without any modification and/or filtering?

File: doc/user-guide/Push-notifications.md [hunk 8]
```diff
@@ -74,17 +74,32 @@ Also, it's important to note, that the first node type on the `plugins` list, wi
...
+   [{server, "https://localhost:8443"}]}
+ ]
+}.
+```
 
-{mod_push_service_mongoosepush, [
-    {pool_name, mongoose_push_http},
-    {api_version, "v2"}
-]}.
+> Please note that there maybe more than one pool in the `outgoing_pools` list.
```
Comment (goddammit) [inline doc/user-guide/Push-notifications.md]: Please note that more than one pool may exist in the `outgoing_pools` list.

Please note that more than one pool may be stored in the `outgoing_pools` list.

Please note that there might be more than one pool in the `outgoing_pools` list.

File: doc/user-guide/Push-notifications.md [hunk 9]
```diff
@@ -74,17 +74,32 @@ Also, it's important to note, that the first node type on the `plugins` list, wi
...
+
+Then add `mod_push_service_mongoosepush` to the modules section in the config file, like below:
```
Comment (goddammit) [inline doc/user-guide/Push-notifications.md]: Then add `mod_push_service_mongoosepush` to the 'modules' section in the config file:

**Comment by mongoose-im** [comment]
[5702.1](https://travis-ci.org/esl/MongooseIM/jobs/436648815) / Erlang 19.3 / small_tests / 6b22d3e30263f3ec28895644766d84c9ed0a98af
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2099/5702/small_tests.19.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2099/5702/small_tests.19.3/small/ct_run.mongooseim@localhost.2018-10-03_14.14.30/lib.mongooseim.logs/run.2018-10-03_14.14.31/suite.log.html)

File: doc/modules/mod_push_service_mongoosepush.md (hunk 7 above)
Comment (michalwski) [reply doc/modules/mod_push_service_mongoosepush.md]: I'd rather stay with the original sentence.

**Pull request merged**